### ISSUE TITLE
mat-menu trigger is setting aria-expanded to null instead of false

### ISSUE THREAD

**Issue by anshul90jain** [angular/components#23704]
#### Reproduction

Use StackBlitz to reproduce your issue: 
- Components: https://stackblitz.com/fork/components-issue
- Harnesses: https://stackblitz.com/fork/harness-issue

Steps to reproduce:
1. Add a button with [matTriggerFor] for a mat menu
`
<button mat-icon-button
        #menuButton
        [matMenuTriggerFor]="exampleMenu">
</button>
`
<mat-menu #exampleMenu="matMenu">
2. Inspect the element in the browser
3. Click on the button to open the menu- it sets the aria-expanded to true
4. Inspect the element when the menu is closed - having it closed does not set it to false (sets it to null)
- https://github.com/angular/components/blob/master/src/material/menu/menu-trigger.ts
- '[attr.aria-expanded]': 'menuOpen || null',
 
 
#### Expected Behavior

When the menu is closed, aria-expanded should be set to false. [WAI-ARIA doc on role="combobox"](https://www.w3.org/TR/wai-aria-practices-1.1/#wai-aria-roles-states-and-properties-6) suggests that aria-expanded is required regardless of whether aria-haspopup is set.

#### Actual Behavior

When the menu is closed, aria-expanded is not set to false, it is set to null

#### Environment

- Angular: Most recent versions
- CDK/Material: Most recent versions
- Browser(s): Chrome
- Operating System (e.g. Windows, macOS, Ubuntu): Ubuntu

This issue is a duplicate of [#16859](https://github.com/angular/components/issues/16859). With the additional note that according to the [WAI-ARIA doc on role="combobox"](https://www.w3.org/TR/wai-aria-practices-1.1/#wai-aria-roles-states-and-properties-6) aria-expanded is required regardless of whether aria-haspopup is set.
